### PR TITLE
WS add billing state methods to sub

### PR DESCRIPTION
**Pull request by will-sauer**
will tell you if sub is in grace period or not
related to: https://legalshield.atlassian.net/secure/RapidBoard.jspa?rapidView=339&modal=detail&selectedIssue=AM01-1093

### CODE REVIEW

File: lib/cashbox/model/subscription.rb [hunk 1]
```diff
@@ -4,6 +4,7 @@ class Subscription < Model
     include Rest::ReadWrite
     include Rest::Cancel
     include Rest::Disentitle
+    GRACE_STATUSES = %w[In\ Retry Grace\ Period Failed\ To\ Collect].freeze
```
Comment (sgelbart) [inline lib/cashbox/model/subscription.rb]: Anyone else think there should be some exceptions to using %w syntax when you have lots of spaces?? ['In Retry', 'Grace Period', 'Failed To Collect']

File: lib/cashbox/model/subscription.rb [hunk 2]
```diff
@@ -31,6 +32,10 @@ class Subscription < Model
     property :starts, coerce: Cashbox::Type.DateTime
     property :status
 
+    def grace?
```
[the remark this answers is not in the document]
Comment (will-sauer) [reply lib/cashbox/model/subscription.rb]: we definitely use / need those. easy add. 

fwiw, we have also built those methods in adonis (they use the sub statuses - which should be in sync with the vindicia sub status). i'll add ^^ to this pr, but do you foresee any risks with having multiple versions of those methods - one in cashbox and one in adonis? i.e. if we add this, do we need to refactor out the same methods on the adonis side?

File: spec/unit/model/subscription_spec.rb [hunk 3]
```diff
@@ -34,6 +34,62 @@
...
+
+    it "GRACE_PERIOD is assigned to the correct value" do
+      expect(Cashbox::Subscription::GRACE_PERIOD).to eql("Grace Period")
+    end
+  end
+  
+  context "billing state methods" do
+    describe "in_retry?" do
+      let(:subscription_retry) { Cashbox::Subscription.new(billing_state: Cashbox::Subscription::IN_RETRY) }
+      let(:subscription_not_retry) { Cashbox::Subscription.new(bililng_state: "Not") }
```
Comment (justinpaulson) [inline spec/unit/model/subscription_spec.rb]: Key is misspelled... in a few places

File: spec/unit/model/subscription_spec.rb [hunk 4]
```diff
@@ -34,6 +34,62 @@
...
+    end
+
+    it "GRACE_PERIOD is assigned to the correct value" do
+      expect(Cashbox::Subscription::GRACE_PERIOD).to eql("Grace Period")
+    end
+  end
+  
+  context "billing state methods" do
+    describe "in_retry?" do
+      let(:subscription_retry) { Cashbox::Subscription.new(billing_state: Cashbox::Subscription::IN_RETRY) }
```
Comment (justinpaulson) [inline spec/unit/model/subscription_spec.rb]: I might hard-code these rather than using the constant that you know they are also checking against.

Comment (will-sauer) [reply spec/unit/model/subscription_spec.rb]: @justinpaulson

Comment (justinpaulson) [reply spec/unit/model/subscription_spec.rb]: Good points, agreed

**Pull request merged**